frontend/src/app/modules/principal/principal-renderer.service.ts

```diff
@@ -108,11 +109,17 @@ export class PrincipalRendererService {
     return fallback;
   }
 
-  private renderUserAvatar(principal:PrincipalLike, fallback:HTMLElement, options:AvatarOptions) {
+  private renderUserAvatar(principal:PrincipalLike, fallback:HTMLElement, options:AvatarOptions):void {
+    const url = this.userAvatarUrl(principal);
+
+    if (!url) {
+      return;
+    }
+
     const image = new Image();
     image.classList.add('op-avatar');
     image.classList.add(`op-avatar_${options.size}`);
-    image.src = this.apiV3Service.users.id(principal.id || '').avatar.toString();
+    image.src = url;
```

Comment:
maybe we only want to instantiate the `Image` if there is a url?

Reply:
We do, there's a guarding if above